test/functional/p2p_erlay.py

```diff
@@ -0,0 +1,596 @@
+#!/usr/bin/env python3
+# Copyhigh (c) 2016-2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test reconciliation-based transaction relay protocol.
+
+"""
+
+from test_framework.p2p import P2PDataStore, p2p_lock
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, hex_str_to_bytes
+from test_framework.messages import (
+    msg_inv, msg_getdata,
+    msg_sendrecon, msg_reqrecon,
+    msg_sketch, msg_reqsketchext, msg_reconcildiff,
+    CTransaction, CInv
+)
+from test_framework.siphash import siphash256
+from enum import IntEnum
+from io import BytesIO
+import random
+import hashlib
+import time
+
+MASK64 = 0xffffffffffffffff
+MSG_WTX = 5
+
+# These parameters are specified in the BIP-0330.
+Q_PRECISION = 2 << 12
+FIELD_BITS = 32
+FIELD_MODULUS = (1 << FIELD_BITS) + 0b10001101
+
+# These parameters are suggested by the Erlay paper based on analysis and simulations.
+DEFAULT_Q = 0.02
+RECON_REQUEST_INTERVAL = 16
+INVENTORY_BROADCAST_INTERVAL = 2
+PENALIZED_INVENTORY_BROADCAST_INTERVAL = INVENTORY_BROADCAST_INTERVAL * 2
+MAX_OUTBOUND_FLOOD_TO = 8
+
+BYTES_PER_SKETCH_CAPACITY = FIELD_BITS / 8
+
+def mul2(x):
+    """Compute 2*x in GF(2^FIELD_BITS)"""
+    return (x << 1) ^ (FIELD_MODULUS if x.bit_length() >= FIELD_BITS else 0)
+
+def mul(x, y):
```

Comment:
I've PR'ed https://github.com/sipa/minisketch/pull/26 which adds a pure Python (slow) full reimplementation of minisketch. You may want to use that instead (it also supports decoding).